packages/flutter/test/material/chip_theme_test.dart

```diff
@@ -219,47 +265,101 @@ void main() {
 
     final RenderBox materialBox = getMaterialBox(tester);
     expect(materialBox, paints..rect(color: chipTheme.backgroundColor));
-    expect(tester.getSize(find.byType(RawChip)), const Size(250, 250)); // label + padding + labelPadding
+    expect(tester.getSize(find.byType(RawChip)), const Size(250, 250)); // icon + label + padding + labelPadding
     expect(getMaterial(tester).elevation, chipTheme.elevation);
     expect(getMaterial(tester).shape, chipTheme.shape);
     expect(getLabelStyle(tester).style.fontSize, 32);
   });
 
+  testWidgets('Chip with Avatar Icon custom iconTheme and uses ChipTheme', (WidgetTester tester) async {
+    const IconThemeData shadowedChipThemeIconTheme = IconThemeData(color: Color(0xffbbeeff));
+    const ChipThemeData chipTheme = ChipThemeData(
+      backgroundColor: Color(0xff112233),
+      elevation: 4,
+      padding: EdgeInsets.all(50),
+      labelPadding: EdgeInsets.all(25),
+      labelStyle: TextStyle(fontSize: 32),
+      shape: RoundedRectangleBorder(),
+      iconTheme: shadowedChipThemeIconTheme,
+    );
+
+    const IconThemeData shadowedCustomIconTheme = IconThemeData(color: Color(0xff332211));
+    const ChipThemeData shadowedChipTheme = ChipThemeData(
+      backgroundColor: Color(0xff332211),
+      elevation: 3,
+      padding: EdgeInsets.all(5),
+      labelPadding: EdgeInsets.all(10),
+      labelStyle: TextStyle(fontSize: 64),
+      shape: CircleBorder(),
+      iconTheme: shadowedCustomIconTheme,
+    );
+
+    const IconThemeData customIconTheme = IconThemeData(color: Color(0xff112233));
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData.light().copyWith(
+          chipTheme: shadowedChipTheme,
+        ),
+        home: ChipTheme(
+          data: chipTheme,
+          child: Builder(
+            builder: (BuildContext context) {
+              return const Material(
+                child: Center(
+                  child: Chip(
+                    label: SizedBox(width: 100, height: 100),
+                    avatar: Icon(Icons.ac_unit),
+                    iconTheme: customIconTheme,
+                  ),
+                ),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
+    expect(getIconTheme(tester), customIconTheme, reason:'Chip Icon Theme does not reflect custom iconTheme.');
+  });
+
```

Comment:
Looks like this test has a confusing title and bloated themes.  We're only if the theme is overridden by `Chip.iconTheme`

Please replace this with the following:

```dart
  testWidgets('ChipTheme.iconTheme overrides local ChipTheme', (WidgetTester tester) async {
    const IconThemeData iconTheme = IconThemeData(color: Color(0xffbbee12));
    await tester.pumpWidget(
      const MaterialApp(
        home: ChipTheme(
          data: ChipThemeData(
            iconTheme: IconThemeData(color: Color(0xffbbeeff)),
          ),
          child: Material(
            child: Center(
              child: Chip(
                label: SizedBox(width: 100, height: 100),
                avatar: Icon(Icons.ac_unit),
                iconTheme: iconTheme,
              ),
            ),
          ),
        ),
      ),
    );

    expect(getIconTheme(tester), iconTheme);
  });
```